### PR TITLE
Updates for v23.06 release

### PR DESCRIPTION
**Pull request by raydouglass**
Updates the required files for the v23.06 release.

I added `previous_releases.json` which contains the schedule for the previous releases allowing easier copy-paste from `releases.json` and rendering on the schedule page. Open to feedback on improving this.

~Also, disabled docs for cudf-java until I get a chance to upload them later today.~ Edit: I've uploaded the javadocs

### CODE REVIEW

**Comment by jarmak-nv** [comment]
Not an approver, but checked out the updated maintainers pages and it looks good to me

File: _data/releases.json
Comment (bdice) [inline _data/releases.json:85]: JSON doesn't permit trailing commas.
```suggestion
  }
```

**Review by bdice — APPROVED** [review]
Approving with one fix.

File: _data/releases.json
Comment (raydouglass) [reply _data/releases.json]: Fixed. I'm surprised my local jekyll nor Netlify complained!

File: releases/schedule.md
Comment (ajschmidt8) [inline releases/schedule.md:1]: there's a discrepancy on this page now where some releases show the date, but others don't.

see the preview here: https://deploy-preview-404--docs-rapids-ai.netlify.app/releases/schedule/

**Review by ajschmidt8 — APPROVED** [review]
approved with a comment.

it can be addressed in a follow-up PR.

the release dates JSON data may eventually need to be re-thought when we merge the docs and main site. for now, this method seems fine.

**Comment by bdice** [comment]
We probably need to delete CLX from the API docs since the "legacy" version is no longer available (?). I'll file a PR for that, go ahead and merge this one.

File: releases/schedule.md
Comment (raydouglass) [reply releases/schedule.md:1]: Yeah, unfortunately, the releases prior to 23.06 were hardcoded tables without the year because they were copy-pasted from the maintainers page which rendered without the year. When I find some time, I'll go back through the old schedule and add the years.

**Pull request merged**
